File: python/5_kyu/string_incrementer.py
# String incrementer
# https://www.codewars.com/kata/54a91a4883a7de5d7800009c/train/python

# Rules:
# 1. if the string is empty, return 1
# 2. if the string does not end with a number, append 1 and return the new string
# 3. if the string ends with a number, incrememt the number by 1

def incrememt_string(s):
    
    # Rule 1
    if len(s) == 0:
        return 1
    
    # Rule 2
    s = list(s)
    if s[-1].isalpha():
        s.append("1")
        return "".join(s)
    
    # Rule 3
    


print(incrememt_string("foo"))
print(incrememt_string("foobar001"))
print(incrememt_string("foobar99"))
print(incrememt_string("foobar099"))
print(incrememt_string(""))

File: javascript/5_kyu/string_incrementer.js
// String incrementer
// https://www.codewars.com/kata/54a91a4883a7de5d7800009c/train/javascript

function incrementString(s) {
    
    // if string is empty, return "1"
    if (s.length == 0) {
        return "1";
    }

    // if the last letter is a letter, return the string itself + 1
    s = s.split("");
    if (s[s.length - 1].match(/[a-zA-Z]+/g)) {
        s.push("1");
        return s;
    }
}

console.log(incrementString("foo"));
// console.log(increment_string("foobar001"));
// console.log(increment_string("foobar99"));
// console.log(increment_string("foobar099"));
// console.log(increment_string(""));
// console.log(increment_string("1s]X>016507.Wiw=T3y67581487bNqji_*C8190579O RA7663Z2Qi~O896nU~/489027691861674357bcT304345054"));